### PR TITLE
feat(`x/marketmap`): module definition

### PR DESCRIPTION
**Pull request by aljo242**
Closes BLO-1035

### CODE REVIEW

File: proto/slinky/mm2/module/v1/module.proto [hunk 1]
```diff
@@ -0,0 +1,21 @@
+syntax = "proto3";
+
+package slinky.mm2.module.v1;
+
+import "cosmos/app/v1alpha1/module.proto";
+
+// Module is the config object of the builder module.
+message Module {
+  option (cosmos.app.v1alpha1.module) = {
+    go_import : "github.com/skip-mev/slinky/x/mm2"
```
Comment (Eric-Warehime) [inline proto/slinky/mm2/module/v1/module.proto:10]: Is the plan to `s/mm2/marketmap/g` after we merge everything?

File: x/mm2/module.go [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+		appmodule.Invoke(InvokeSetMarketMapHooks),
+	)
+}
+
+// Inputs contains the dependencies required for module construction.
+type Inputs struct {
+	depinject.In
+
+	// module dependencies
+	Config       *marketmapmodulev1.Module
```
Comment (Eric-Warehime) [inline x/mm2/module.go:189]: We're using the existing marketmap to construct the new marketmap? Is that correct?

File: proto/slinky/mm2/module/v1/module.proto (hunk 1 above)
Comment (aljo242) [reply proto/slinky/mm2/module/v1/module.proto:10]: yes.

File: x/mm2/module.go (hunk 2 above)
Comment (aljo242) [reply x/mm2/module.go:189]: no - if you look at the import path - this is also from the new proto

Comment (Eric-Warehime) [reply x/mm2/module.go:189]: Ah I see. Just a confusing import name I guess. :O

**Review by Eric-Warehime — APPROVED** [review]

**Review by davidterpay — APPROVED** [review]

**Pull request merged**